kafka.bootstrap.servers can't handle YAML list

The documentation states that you can specify a list of Kafka servers with `kafka.bootstrap.servers`.

https://micronaut-projects.github.io/micronaut-kafka/latest/guide/index.html#kafkaQuickStart

But doing so results in an error. The list is converted to a string but Kafka expects a list.

A workaround is to specify a commma separated string like this:
```
kafka:
    bootstrap:
        servers: kafka1:9092, kafka2:9092, kafka3:9092
```
### Steps to Reproduce

Create a micronaut application and add Kafka dependency.

`compile 'io.micronaut.configuration:micronaut-kafka'`

Then add a list of servers (one list entry is enough) to applicationn.yml.

```
kafka:
    bootstrap:
        servers:
            - localhost:9092
```

Try to start the application.

`./gradlew run`

### Expected Behaviour

The application should start and try to connect to a Kafka broker on localhost.

### Actual Behaviour

```
Caused by: org.apache.kafka.common.config.ConfigException: Invalid url in bootstrap.servers: [localhost:9092]
        at org.apache.kafka.clients.ClientUtils.parseAndValidateAddresses(ClientUtils.java:56)
        at org.apache.kafka.clients.consumer.KafkaConsumer.<init>(KafkaConsumer.java:712)
        ... 26 common frames omitted
```

The logging output shows the invalid config string:
```
INFO  o.a.k.c.consumer.ConsumerConfig - ConsumerConfig values: 
        auto.commit.interval.ms = 5000
        auto.offset.reset = latest
        bootstrap.servers = [[localhost:9092]]
```

### Environment Information

- **Operating System**: MacOS
- **Micronaut Version:** 1.1.0.M2
- **JDK Version:** 1.8.0_162

### Example Application

https://github.com/goeh/micronaut-kafka-issue-26

--- a/kafka/src/main/java/io/micronaut/configuration/kafka/config/KafkaDefaultConfiguration.java
+++ b/kafka/src/main/java/io/micronaut/configuration/kafka/config/KafkaDefaultConfiguration.java
@@ -19,7 +19,6 @@ import io.micronaut.context.annotation.ConfigurationProperties;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.env.Environment;
 import io.micronaut.core.convert.ConversionService;
-import io.micronaut.core.convert.DefaultConversionService;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 
 import java.time.Duration;
@@ -44,7 +43,6 @@ public class KafkaDefaultConfiguration extends AbstractKafkaConfiguration {
     @SuppressWarnings("WeakerAccess")
     public static final int DEFAULT_HEALTHTIMEOUT = 10;
 
-    private static final ConversionService CONVERSION_SERVICE = new DefaultConversionService();
     private Duration healthTimeout = Duration.ofSeconds(DEFAULT_HEALTHTIMEOUT);
 
     /**
@@ -88,8 +86,8 @@ public class KafkaDefaultConfiguration extends AbstractKafkaConfiguration {
             return !Stream.of("embedded", "consumers", "producers", "streams").anyMatch(key::startsWith);
         }).forEach(entry -> {
             Object value = entry.getValue();
-            if (CONVERSION_SERVICE.canConvert(entry.getValue().getClass(), String.class)) {
-                Optional<Object> converted = CONVERSION_SERVICE.convert(entry.getValue(), String.class);
+            if (ConversionService.SHARED.canConvert(entry.getValue().getClass(), String.class)) {
+                Optional<?> converted = ConversionService.SHARED.convert(entry.getValue(), String.class);
                 if (converted.isPresent()){
                     value = converted.get();
                 }

--- a/kafka/src/main/java/io/micronaut/configuration/kafka/config/KafkaDefaultConfiguration.java
+++ b/kafka/src/main/java/io/micronaut/configuration/kafka/config/KafkaDefaultConfiguration.java
@@ -18,9 +18,13 @@ package io.micronaut.configuration.kafka.config;
 import io.micronaut.context.annotation.ConfigurationProperties;
 import io.micronaut.context.annotation.Requires;
 import io.micronaut.context.env.Environment;
+import io.micronaut.core.convert.ConversionService;
+import io.micronaut.core.convert.DefaultConversionService;
 import org.apache.kafka.clients.consumer.ConsumerConfig;
 
 import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
 import java.util.Properties;
 import java.util.stream.Stream;
 
@@ -40,6 +44,7 @@ public class KafkaDefaultConfiguration extends AbstractKafkaConfiguration {
     @SuppressWarnings("WeakerAccess")
     public static final int DEFAULT_HEALTHTIMEOUT = 10;
 
+    private static final ConversionService CONVERSION_SERVICE = new DefaultConversionService();
     private Duration healthTimeout = Duration.ofSeconds(DEFAULT_HEALTHTIMEOUT);
 
     /**
@@ -76,12 +81,22 @@ public class KafkaDefaultConfiguration extends AbstractKafkaConfiguration {
     }
 
     private static Properties resolveDefaultConfiguration(Environment environment) {
-        Properties values = environment.getProperty(PREFIX, Properties.class).orElseGet(Properties::new);
+        Map<String, Object> values = environment.getProperties(PREFIX);
         Properties properties = new Properties();
         values.entrySet().stream().filter(entry -> {
             String key = entry.getKey().toString();
             return !Stream.of("embedded", "consumers", "producers", "streams").anyMatch(key::startsWith);
-        }).forEach(entry -> properties.put(entry.getKey(), entry.getValue()));
+        }).forEach(entry -> {
+            Object value = entry.getValue();
+            if (CONVERSION_SERVICE.canConvert(entry.getValue().getClass(), String.class)) {
+                Optional<Object> converted = CONVERSION_SERVICE.convert(entry.getValue(), String.class);
+                if (converted.isPresent()){
+                    value = converted.get();
+                }
+            }
+            properties.setProperty(entry.getKey(), value.toString());
+
+        });
         return properties;
     }
 }

--- a/kafka/src/main/java/io/micronaut/configuration/kafka/metrics/AbstractKafkaMetricsReporter.java
+++ b/kafka/src/main/java/io/micronaut/configuration/kafka/metrics/AbstractKafkaMetricsReporter.java
@@ -44,7 +44,7 @@ abstract class AbstractKafkaMetricsReporter implements MetricsReporter, MeterBin
 
     @Override
     public void bindTo(@NonNull MeterRegistry registry) {
-        if(!METER_REGISTRIES.contains(registry)) {
+        if (!METER_REGISTRIES.contains(registry)) {
             METER_REGISTRIES.add(registry);
         }
     }

--- a/kafka/src/test/groovy/io/micronaut/configuration/kafka/KafkaConfigurationSpec.groovy
+++ b/kafka/src/test/groovy/io/micronaut/configuration/kafka/KafkaConfigurationSpec.groovy
@@ -118,4 +118,34 @@ class KafkaConfigurationSpec extends Specification {
         consumer.close()
         applicationContext.close()
     }
+
+    void "test configure list fields default properties"() {
+        given:
+        ApplicationContext applicationContext = ApplicationContext.run(
+                ("kafka.${ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG}".toString()):Arrays.asList("localhost:1111","localhost:1112"),
+                ("kafka.${ConsumerConfig.GROUP_ID_CONFIG}".toString()):"mygroup",
+                ("kafka.consumers.default." + ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG): StringDeserializer.name,
+                ("kafka.consumers.default." + ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG): StringDeserializer.name
+        )
+
+        when:
+        AbstractKafkaConsumerConfiguration config = applicationContext.getBean(AbstractKafkaConsumerConfiguration)
+        Properties props = config.getConfig()
+
+        then:
+        props[ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG] == "localhost:1111,localhost:1112"
+        props[ConsumerConfig.GROUP_ID_CONFIG] == "mygroup"
+
+        when:
+        Consumer consumer = applicationContext.createBean(Consumer, config)
+
+        then:
+        consumer != null
+
+        cleanup:
+        consumer.close()
+        applicationContext.close()
+
+
+    }
 }

